src/entity/posts/utils.ts

```diff
@@ -344,20 +348,75 @@ export const addNewPost = async (
   });
 };
 
-export const createSharePost = async (
-  con: DataSource | EntityManager,
-  sourceId: string,
-  userId: string,
-  postId: string,
-  commentary: string,
-): Promise<SharePost> => {
+const validateCommentary = async (commentary: string) => {
   if (commentary.length > MAX_COMMENTARY_LENGTH) {
     throw new ValidationError(
       JSON.stringify({
         commentary: `max size is ${MAX_COMMENTARY_LENGTH} chars`,
       }),
     );
   }
+  return true;
+};
+
+export const createPrivatePost = async (
+  con: DataSource,
+  logger: EventLogger,
+  sourceId: string,
+  userId: string,
+  url: string,
+  commentary: string,
+): Promise<void> => {
+  await validateCommentary(commentary);
+  const id = await generateShortId();
+  const sharedId = await generateShortId();
+
+  return con.transaction(async (entityManager) => {
+    try {
+      await entityManager.getRepository(ArticlePost).insert({
+        id,
+        shortId: id,
+        createdAt: new Date(),
+        sourceId,
+        url,
```

Comment:
I think we should set `canonicalUrl` as well here. Not highly important but just to make sure it's not empty